### DISCUSSION TITLE
[DOC] input formats for transformations

### DISCUSSION THREAD

**Discussion by michaelkalz** [alan-turing-institute/sktime#4387] [Q&A]
#### Describe the issue linked to the documentation

I am a medical student with some background in CS and I am trying to build a new API to utilise the transformers for our machine learning pipeline tool (https://photon-ai.com/). However, I am having quite a rough time trying to get the transformers running on some of our data, that come in the shape: (rat instance, sensors, measurements) and need to be np.arrays.

There are not that many transformers with examples/example data and (at least to me) it is not clear when to use pd.Df or np.array as input and in which format (input and output) without digging deep(er) into the linked papers (if linked) or github repo.

#### Suggest a potential alternative/fix

Could you please provide some more information about input/output type and format and/or more examples for the transformation section?

**Comment by fkiraly** [comment]
Have you seen the transformations tutorial?

Tutorial 3, e.g., here on binder:
https://mybinder.org/v2/gh/sktime/sktime/main?filepath=examples

There's also an extended tutorial on transformations and pipelines here:
https://github.com/sktime/sktime-tutorial-pydata-global-2022

**Comment by fkiraly** [comment]
> that come in the shape: (rat instance, sensors, measurements) and need to be np.arrays.

This sounds like it is already in the `numpy3D` format (mtype) accepted by all `sktime` transformers and should work out of the box with all transformers.

>  it is not clear when to use pd.Df or np.array as input and in which format (input and output) without digging deep(er) into the linked papers (if linked) or github repo.

Both input formats - `pd-multiindex` and `numpy3D` should work with all transformers!

The papers are outdated, the online documentation and the tutorial notebooks on the main repo are "living" and should be evergreen.

**Reply by jernsting** [reply]
But if this is `numpy3D` Format, how would I then expect the target? My assumption would be, that there is one label for the whole instance, but this seems to be different from what sktime expects. [instance given our input (instances, sensors, measurements)]

**Reply by jernsting** [reply]
In other words: The `numpy3D` input is not usable for multivariate input (ref https://sktime.readthedocs.io/en/latest/examples/AA_datatypes_and_datasets.html#Section-1.1.3:-Time-series---the-%22np.ndarray%22-mtype, capabilities: cannot represent multivariate series). 

However, it seems that Panels are actually a generalisation of multivariate input? So using the Panel input type with data from just one instance should be equal to using multivariate input?

**Comment by fkiraly** [comment]
Just noticed that we haven't yet moved the pydata 2022 material to the sktime main repo and it might be a bit patchy... opened an issue here
https://github.com/sktime/sktime/issues/4373

In the meanwhile, I hope the link to the pydata 2022 material helps.

**Comment by fkiraly** [comment]
moving this to the discussion forum

**Comment by jernsting** [comment]
Documentation and repository are out of sync. E.g. https://github.com/sktime/sktime/blob/v0.16.1/sktime/transformations/series/fourier.py clearly does not contain FourierTransform, but latest documentation (https://www.sktime.net/en/latest/api_reference/auto_generated/sktime.transformations.series.fourier.FourierTransform.html) states that it should exist.

Also version specific links in the documentation do not work. For example the link to the documentation for v0.16.1 does not work: https://sktime.readthedocs.io/en/v0.16.1/

**Reply by fkiraly** [reply]
Thanks for bringing this up!

Discussion moved to two issues:

* https://github.com/sktime/sktime/issues/4407
* https://github.com/sktime/sktime/issues/4408
